ton_client/src/tvm/types.rs

```diff
@@ -72,3 +74,74 @@ impl ResolvedExecutionOptions {
         })
     }
 }
+
+pub async fn resolve_blockchain_config(
+    context: &Arc<ClientContext>,
+    provided_config: Option<String>,
+) -> ClientResult<Arc<BlockchainConfig>> {
+    if let Some(config) = provided_config {
+        blockchain_config_from_boc(context, &config).await.map(Arc::new)
+    } else {
+        get_default_config(context).await
+    }
+}
+
+pub(crate) async fn get_default_config(context: &Arc<ClientContext>) -> ClientResult<Arc<BlockchainConfig>> {
+    if let Some(config) = &*context.blockchain_config.read().await {
+        return Ok(config.clone());
+    }
+
+    let mut config_lock = context.blockchain_config.write().await;
+    if let Some(config) = &*config_lock {
+        return Ok(config.clone());
+    }
+
+    let config = if let Ok(link) = context.get_server_link() {
+        get_network_config(link)
+            .await?
+            //.unwrap_or_default()
+    } else {
+        Default::default()
```

Comment:
I think we mustn't cache default config in context when get_server_link has failed.
I suggest `return Ok(Default::default())`